src/node/rpc/common_endpoint_registry.h

```diff
@@ -7,31 +7,174 @@
 #include "http/ws_consts.h"
 #include "json_handler.h"
 #include "node/code_id.h"
+#include "node/quote.h"
+#include "node/rpc/node_interface.h"
 
 namespace ccf
 {
+  struct Quote
+  {
+    NodeId node_id = {};
+    std::string raw = {}; // < Hex-encoded
+
+    std::string error = {};
+    std::string mrenclave = {}; // < Hex-encoded
```

Comment:
Can we make those doxygen comments? That way they should show up when we expose the type in the doc.

Reply:
I've changed the API here. `get_quote_for_this_node_v1` just returns the raw quote as `std::vector<uint8_t>` (with a `QuoteFormat` enum of `oe_sgx_v1`). Then this `Quote` struct only lives in the `node_frontend`, as part of our HTTP API, but not as part of the supported C++ API. Separately I've also remove the `error` field from this struct in its new home, I think it was an odd way of indicating quote errors.

Reply:
But I will add doxygen comments on these methods, since we want them to be well documented. I'll add a TODO list...